### PR TITLE
Query utxo by output reference

### PR DESCRIPTION
**Pull request by rynoV**
Adds `getUtxo` function implementing [ogmios' utxo by txin query](https://ogmios.dev/mini-protocols/local-state-query/#utxo-by-txin).

### Pre-review checklist

- [x] All code has been formatted using our config (`make format` for Purescript, `nixpkgs-fmt` for Nix)
- [x] All Purescript imports are explicit, including constructors
- [x] **All** existing examples have been run locally against a fully-synced testnet node
- [x] The integration and unit tests have been run (`npm run test`) locally
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`)

### CODE REVIEW

File: src/QueryM/Utxos.purs [hunk 1]
```diff
@@ -42,26 +43,33 @@ import Wallet (Wallet(Gero, Nami, KeyWallet))
 -- | Gets utxos at an (internal) `Address` in terms of (internal) `Cardano.Transaction.Types`.
 -- | Results may vary depending on `Wallet` type.
 utxosAt :: Address -> QueryM (Maybe UtxoM)
-utxosAt addr = asks _.wallet >>= maybe (allUtxosAt addr) (utxosAtByWallet addr)
+utxosAt = mkUtxoQuery
+  <<< mkOgmiosRequest Ogmios.queryUtxosAtCall _.utxo
+  <<< addressToOgmiosAddress
+
+-- | `getUtxo ref` queries for the utxo given by `ref`.
```
[the remark this answers is not in the document]
Comment (klntsky) [reply src/QueryM/Utxos.purs:50]: `Queries for UTxO given a transaction input`

File: src/Contract/Utxos.purs [hunk 2]
```diff
@@ -38,6 +40,19 @@ utxosAt address = do
     )
     mCardanoUtxos
 
+-- | `getUtxo ref` queries for the utxo given by `ref`.
```
Comment (klntsky) [inline src/Contract/Utxos.purs:43]: Same as below

File: src/QueryM/Ogmios.purs [hunk 3]
```diff
@@ -170,14 +168,24 @@ queryUtxosCall = mkOgmiosCallType
...
 
+-- | Queries Ogmios for the utxo with the given output reference.
+queryUtxoCall :: JsonWspCall TransactionInput UtxoQR
+queryUtxoCall = mkOgmiosCallType
+  { methodname: "Query"
+  , args: { query: _ } <<< { utxo: _ } <<< singleton <<< renameFields <<< unwrap
+  }
+  Proxy
+  where
+  renameFields { transactionId, index } = { txId: transactionId, index }
```
Comment (vvtran) [inline src/QueryM/Ogmios.purs:187]: We tend to have explicit type sigs inside `where` even if obvious.

**Review by firefrorefiddle — APPROVED** [review]

File: src/Contract/Utxos.purs [hunk 4]
```diff
@@ -2,31 +2,43 @@
...
 -- | on wallet variance.
+-- |
+-- | NOTE: Querying for UTxOs by address is deprecated. See
+-- | [here](https://github.com/CardanoSolutions/ogmios/issues/199).
 utxosAt
-  :: forall (r :: Row Type). Address -> Contract r (Maybe Transaction.UtxoM)
+  :: forall (r :: Row Type)
+   . Warn
+       ( Text
+           "`utxosAt`: Querying for UTxOs by address is deprecated. See https://github.com/CardanoSolutions/ogmios/issues/199."
```
Comment (klntsky) [inline src/Contract/Utxos.purs:38]: Please add this link instead: https://github.com/Plutonomicon/cardano-transaction-lib/issues/536

I think it's better to point users to our issue, to make it clear how we are going to handle that problem in the future.

**Review by klntsky — CHANGES_REQUESTED** [review]

**Review by jy14898 — CHANGES_REQUESTED** [review]

**Comment by klntsky** [comment]
Looks good. Please resolve the conflict.

**Review by klntsky — APPROVED** [review]

**Pull request merged**